src/batch-submitter/batch-submitter.ts

```diff
@@ -141,66 +151,52 @@ export abstract class BatchSubmitter {
   }
 
   public static async getReceiptWithResubmission(
-    response: TransactionResponse,
-    receiptPromises: Array<Promise<TransactionReceipt>>,
-    signer: Signer,
-    numConfirmations: number,
-    resubmissionTimeout: number,
+    txFunc: (gasPrice) => Promise<TransactionReceipt>,
+    resubmissionConfig: ResubmissionConfig,
     log: Logger,
   ): Promise<TransactionReceipt> {
-    const receiptPromise = response.wait(numConfirmations)
-    receiptPromises.push(receiptPromise)
-    // Wait for the tx & if it takes too long resubmit
-    const sleepAndReturnResubmit = async (timeout: number) => {
-      await new Promise((r) => setTimeout(r, timeout))
-      return 'resubmit'
-    }
-    const promises = [...receiptPromises, sleepAndReturnResubmit(
-      resubmissionTimeout
-    )]
-    const val = await bPromise.any(promises)
-
-    if (val === 'resubmit') {
-      log.debug(`Tx resubmission timeout reached for hash: ${response.hash}; nonce: ${response.nonce}.
-                Resubmitting tx...`)
-      const tx = {
-        to: response.to,
-        data: response.data,
-        nonce: response.nonce,
-        value: response.value,
-        gasLimit: response.gasLimit,
-      }
-      const newRes = await signer.sendTransaction(tx)
-      log.debug('Resubmission tx response:', newRes)
-      return BatchSubmitter.getReceiptWithResubmission(
-        newRes,
-        receiptPromises,
-        signer,
-        numConfirmations,
-        resubmissionTimeout,
-        log
-      )
-    } else {
-      return val
-    }
+    const {
+      resubmissionTimeout,
+      minGasPriceInGwei,
+      maxGasPriceInGwei,
+      gasRetryIncrement
+    } = resubmissionConfig
+
+    const receipt = await ynatm.send({
+        sendTransactionFunction: txFunc,
+        minGasPrice: ynatm.toGwei(minGasPriceInGwei),
+        // minGasPriceInGwei,
```

Comment:
These comments are dead code right?

Reply:
yes!! i'll re-review this myself and cleanup ugh